packages/create-app/src/lib/tasks.ts

```diff
@@ -199,9 +199,21 @@ export async function createTemporaryAppFolderTask(tempDir: string) {
  * @param appDir - location of application to build
  */
 export async function buildAppTask(appDir: string) {
+  process.chdir(appDir);
+
+  await Task.forItem('determining', 'yarn version', async () => {
+    const result = await exec('yarn --version');
+    const yarnVersion = result.stdout?.trim();
+
+    if (yarnVersion && !yarnVersion.startsWith('1.')) {
+      throw new Error(
+        `@backstage/create-app requires Yarn v1, found '${yarnVersion}'. You can migrate the project to Yarn 3 after creation using https://backstage.io/docs/tutorials/yarn-migration`,
```

Comment:
I think it already works by running `yarn set version classic`

Reply:
Are you suggesting we run that command here instead of erroring out?

Think that might make sense. I guess my main concern is that you end up with projects looking different depending on the configuration of the machine of the project creator.

Reply:
Running it or mentioned it in the error both works I think and will help people like me who don't use Yarn and get stuck in hours long attempt to install Yarn v1 :)